lib/datadog/appsec/processor.rb

```diff
@@ -14,9 +14,12 @@ def initialize(processor)
           @time_ext_ns = 0.0
           @timeouts = 0
           @events = []
+          @run_mutex = Mutex.new
         end
 
         def run(input, timeout = WAF::LibDDWAF::DDWAF_RUN_TIMEOUT)
+          @run_mutex.lock
+
           start_ns = Core::Utils::Time.get_time(:nanosecond)
```

Comment:
Minor: Consider using `@run_mutex.synchronize` -- I think that one's a bit easier to maintain and "break" with future changes.

Reply:
Rationale:

- blocks carry a cost: def/ensure/end is faster
- it makes the function return value less explicit (you have to konw/remember/lookup that synchronize does return the block value)
- it creates whitespace churn on the code change
- maybe I'm doing too much Go but def/ensure/end is a common pattern that matches Go's `defer` ;)

Reply:
Added a comment to clarify the intent!